src/site/topic2-cpp.rst

```diff
@@ -0,0 +1,648 @@
+    print(theInput)
+
+
+**Java**
+
+.. code-block:: Java
+    :linenos:
+
+    import java.io.BufferedReader;
+    import java.io.InputStreamReader;
+    import java.io.IOException;
+
+    public class SomeClass {
+        public static void main(String[] args){
+
+            // Create a Stream Reader with the standard input
+            InputStreamReader stream = new InputStreamReader(System.in);
+
+            // Give the Stream Reader to a Buffered Reader
+            BufferedReader reader = new BufferedReader(stream);
+
+            // We use the Buffered Reader to read the actual stream
+            // We use a try & catch because readLine may throw an
+            // exception that we must deal with
+            try {
+                String theLine = reader.readLine();
+                System.out.println(theLine);
+            } catch (IOException e){
+                System.out.println("Something bad happened when reading.");
+            }
+        }
+    }
+
+
+**C++**
+
+
+* Below is an example of inputting a primitive type
+
+.. code-block:: cpp
+    :linenos:
+
+    int a;
+    std::cout << "gimmie something: ";
+    std::cin >> a;
+    std::cout << a << std::endl;
+
+
+* If we want to read two things on a single line, you could do this
+
+.. code-block:: cpp
+    :linenos:
+
+    int a, b;
+    std::cout << "gimmie 2 somethings: ";
+    std::cin >> a >> b;
+    std::cout << a << " and " << b << std::endl;
+
+
+* Reading in a string is a little different, but still relatively simple
+
+.. code-block:: cpp
+    :linenos:
+    :emphasize-lines: 3
+
+    std::string a;
+    std::cout << "gimmie something: ";
+    std::getline(std::cin, a);
+    std::cout << a << std::endl;
+
+
+Functions
+=========
+
+**Python**
+
+.. code-block:: Python
+    :linenos:
+    :emphasize-lines: 2
+
+    # Python --- Declaring a function
+    def someFunction(a, b):
+        c = a + b
+        return c
+
+    # Call the function
+    result = someFunction(1, 2)
+    print(result)
+
+
+**Java**
+
+.. code-block:: Java
+    :linenos:
+    :emphasize-lines: 5, 10
+
+    // Java
+    public class SomeClass {
+        public static void main(String[] args) {
+            // Call the function
+            int result = someFunction(1, 2);
+            System.out.println(result);
+        }
+
+        // Declare the Function
+        static int someFunction(int a, int b) {
+            int c = a + b;
+            return c;
+        }
+    }
+
+
+**C++**
+
+.. code-block:: cpp
+    :linenos:
+    :emphasize-lines: 2, 8
+
+    // Declare the function
+    int someFunction(int a, int b){
+        int c = a + b;
+        return c;
+    }
+
+    int main(){
+        std::cout << someFunction(1, 2) << std::endl;
+    }
+
+* You will see that the Java and C++ examples are very similar
+* The one catch here is that the function is defined before main
+    * We can put it after main if really we want, but then we would need to add a function prototype before the main
```

Comment:
I think function _declaration_ is the more commonly used name. That's at least what I've seen more often; it also fits the "declare before use" thing.